xclim/core/units.py

```diff
@@ -566,12 +567,10 @@ def to_agg_units(
     ...     np.arange(52) + 10,
     ...     dims=("time",),
     ...     coords={"time": time},
-    ...     attrs={"units": "degC"},
     ... )
-    >>> degdays = (
-    ...     (tas - 16).clip(0).sum("time")
-    ... )  # Integral of  temperature above a threshold
-    >>> degdays = to_agg_units(degdays, tas, op="delta_prod")
+    >>> dt = (tas - 16).assign_attrs(units="delta_degC")
+    >>> degdays = dt.clip(0).sum("time")  # Integral of temperature above a threshold
+    >>> degdays = to_agg_units(degdays, dt, op="integral")
```

Comment:
@aulemahal This is a consequence of the reorganizing. We can't be "as" lazy as before and let "integral" know that we mean a temperature difference. We can still be kind of careless with the rest of operations, just have to be careful with the (T-T0) part.

Reply:
I'm thinking of a potential problem: 

Would this kind of computation sometimes happen?
```python
degdays = tas_degC.clip(tas_ref).sum("time") 
```

Essentially, instead of:
```
tas_degC - tas_ref > 0
```
you look at
```
tas_degC  > tas_ref
```

In this last case, it would be weird to define a temperature delta so that things work out.

Is it reasonable to expect users/developers should know: Ok, I want degdays, so this must be a temperature difference?

Reply:
I don't think your example would happen. Summing relative units makes no sense to me!
Otherwise, it seems reasonable to me to expect developers to realize this.